infrastructure/stacks/lambda-iam-roles/main.tf

```diff
@@ -126,3 +126,57 @@ resource "aws_iam_role_policy" "event_sender_policy" {
 }
 EOF
 }
+
+resource "aws_iam_role" "fifo_dlq_handler_role" {
+  name               = var.fifo_dlq_handler_role_name
+  path               = "/"
+  description        = ""
+  assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": "sts:AssumeRole",
+      "Principal": {
+        "Service": "lambda.amazonaws.com"
+      },
+      "Effect": "Allow",
+      "Sid": ""
+    }
+  ]
+}
+EOF
+}
+
+resource "aws_iam_role_policy" "fifo_dlq_handler_policy" {
+  name   = "fifo_dlq_handler_policy"
+  role   = aws_iam_role.fifo_dlq_handler_role.id
+  policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Action": [
+        "logs:CreateLogGroup",
+        "logs:CreateLogStream",
+        "logs:PutLogEvents",
+        "ec2:CreateNetworkInterface",
+        "ec2:DescribeNetworkInterfaces",
+        "ec2:DeleteNetworkInterface",
+        "ec2:AssignPrivateIpAddresses",
+        "ec2:UnassignPrivateIpAddresses",
+        "ec2:DescribeSecurityGroups",
+        "ec2:DescribeSubnets",
+        "ec2:DescribeVpcs",
+        "xray:PutTraceSegments",
+        "xray:PutTelemetryRecords",
+        "sqs:*",
```

Comment:
would be nice if we could be more specific here on both the actions and resources with relation to SQS and DynamoDB, exercising a principal of least privilege

Reply:
Updated to only allow limited permissions on our SQS queues